runtime/vm/jvminit.c

```diff
@@ -2439,6 +2439,14 @@ static UDATA checkArgsConsumed(J9PortLibrary* portLibrary, J9VMInitArgs* j9vm_ar
 		}
 	}
 
+	/* Consuming the shared class options if it is used without -Xshareclasses */
+	findArgInVMArgs( PORTLIB, j9vm_args, EXACT_MATCH, VMOPT_XXSHARECLASSESENABLEBCI, NULL, TRUE);
```

Comment:
This can be optimized, these options only need to be consumed when ignoreUnrecongizedXXColonOptions is false, and when -Xshareclasses is not enabled, for which I think you can check `vm->sharedClassPreinitConfig` as per https://github.com/eclipse/openj9/blob/dde4b86e24f7e01f8f70c8d16bde45b0cb4efdd5/runtime/vm/jvminit.c#L1754

Reply:
Actually, better to check vm->sharedCacheAPI->xShareClassesPresent
https://github.com/eclipse/openj9/blob/c657c0951cc2f5c070ad40efe260d26614cebbe9/runtime/shared/shrclssup.c#L65

Reply:
Changes have been made and tested, please review. Thank you!